azext_edge/edge/providers/orchestration/clone.py

```diff
@@ -0,0 +1,1419 @@
+                resources=self._enumerate_resources(),
+                template_gen=TemplateGen(
+                    self.rcontainer_map, self.parameter_map, self.variable_map, self.metadata_map
+                ),
+                user_assigned_mis=self.instance_identities,
+            )
+
+    def _enumerate_resources(self):
+        enumerated_map: dict = {}
+
+        def __enumerator(rcontainer_map: Dict[str, ResourceContainer]):
+            for resource in rcontainer_map:
+                target_rcontainer = rcontainer_map[resource]
+                if isinstance(target_rcontainer, ResourceContainer):
+                    if "id" not in target_rcontainer.resource_state:
+                        continue
+                    parsed_id = parse_resource_id(target_rcontainer.resource_state["id"].lower())
+                    key = f"{parsed_id['namespace']}/{parsed_id['type']}"
+                    if "resource_type" in parsed_id and parsed_id["resource_type"] != parsed_id["type"]:
+                        key += f"/{parsed_id['resource_type']}"
+
+                    items: list = enumerated_map.get(key, [])
+                    items.append(parsed_id)
+                    enumerated_map[key] = items
+                if isinstance(target_rcontainer, DeploymentContainer):
+                    __enumerator(target_rcontainer.rcontainer_map)
+
+        __enumerator(self.rcontainer_map)
+        return enumerated_map
+
+    def _build_parameters(self):
+        self.parameter_map.update(build_parameter(name=TemplateParams.CLUSTER_NAME.value))
+        self.parameter_map.update(
+            build_parameter(
+                name=TemplateParams.CLUSTER_NAMESPACE.value, default=self.custom_location["properties"]["namespace"]
+            )
+        )
+        self.parameter_map.update(
+            build_parameter(
+                name=TemplateParams.CUSTOM_LOCATION_NAME.value,
+                default=self.custom_location["name"],
+            )
+        )
+        self.parameter_map.update(
+            build_parameter(name=TemplateParams.INSTANCE_NAME.value, default=self.instance_record["name"])
+        )
+        self.parameter_map.update(
+            build_parameter(
+                name=TemplateParams.OPS_EXTENSION_NAME.value,
+                default=(
+                    get_ops_extension_name(self.custom_location["properties"].get("clusterExtensionIds", []))
+                    or "[format('azure-iot-operations-{0}', parameters('resourceSlug'))]"
+                ),
+            )
+        )
+
+        self.parameter_map.update(
+            build_parameter(
+                name=TemplateParams.RESOURCE_SLUG.value,
+                default=(
+                    "[take(uniqueString(resourceGroup().id, "
+                    "parameters('clusterName'), parameters('clusterNamespace')), 5)]"
+                ),
+            )
+        )
+
+    def _build_variables(self):
+        pass
```

Comment:
pass?

Reply:
There used to be template variables, then I simplified and removed them, and this was a placeholder in case variables were to be added in the future.

Reply:
Removed the placeholder, will add later if necessary.